exception.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2021 Roger Chapman and the v8go contributors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package v8go
+
+import (
+	// #include <stdlib.h>
+	// #include "v8go.h"
+	"C"
+
+	"fmt"
+	"unsafe"
+)
+
+// NewRangeError creates a RangeError.
+func NewRangeError(iso *Isolate, msg string) *Exception {
```

Comment:
There is already the JSError type which is used to return exceptions from v8go functions.  It seems like there should be consistency between the exception value returned by other v8go functions and these functions to construct errors.  That should make it easier to propagate exceptions, even without adding a new feature to return errors from function callbacks.

Currently, JSError is converting the underlying JS error object to a string and storing it in the Message field.  Instead, it should store the underlying JS error object, so that the error can be propagated.  That Message field could be deprecated in favour of a method so that the conversion can be done on-demand in the future.  It looks like the stack trace can similarly be obtained on-demand using [v8::Exception::GetStackTrace](https://v8.github.io/api/head/classv8_1_1Exception.html#a47e3613ae924508cc0ca29ad05e2464a).

The JSError.Location field is derived from the a v8::Message on v8::TryCatch, which may be present without a stack trace for compilation errors.  v8::Exception::CreateMessage can be used to obtained a v8::Message from an exception value, but we don't want to automatically create it in a getter method.  Since it is a public field, it will need to continue populating on creation for the current code paths, but it could similarly be deprecated in favour of a method to obtain this on-demand for new code paths.  This on-demand code path could at least obtain the when the error has a stack trace, even if we don't get it to initially work on this new code path errors with a location but no stack trace.

It seems like this refactoring and the addition of these functions to create exceptions could be split out into a separate PR.

Reply:
I've opened this as an issue (https://github.com/rogchap/v8go/issues/274) for greater visibility on this issue with JSError

Reply:
I don't remember exactly why I created a new type for it, but it was probably to avoid having to refactor `JSError` (avoiding merge conflicts). If there's consensus that's better, that sounds good to me. Thanks for reviewing.